sds_in_a_box/SDSCode/download_query_api.py

```diff
@@ -0,0 +1,101 @@
+import boto3
+import botocore
+import os
+import json
+import logging
+
+logger = logging.getLogger()
+logging.basicConfig()
+logger.setLevel(logging.INFO)
+
+
+def http_response(header_type='text/html', status_code=200, body='Success'):
+    """customize http response.
+
+    Args:
+        header_type (str, optional): [description]. Defaults to 'text/html'.
+        status_code (int, optional): [description]. Defaults to 200.
+        body (str, optional): [description]. Defaults to 'Success'.
+
+    Returns:
+        [json]: API response
+    """
+    return  {
+        'headers': {
+            'Content-Type': header_type,
+        },
+        'statusCode': status_code,
+        'body': body
+    }
+
+def lambda_handler(event, context):
+    """This lambda handler checks if this file exists or not. If file doesn't exist, it
+    gives back an error. Otherwise, it returns pre-signed s3 url that user can use to
+    donwload data from s3.
+
+    Args:
+        event (dict): input to lambda
+        context : This is not used.
+    """
+
+    logger.info(event)
+
+    one_day = 86400
+    url_life = os.environ.get("URL_EXPIRE", one_day)
+
+    if  event['rawQueryString'] == '':
+        response_body = '''No input given. It requires bucket and filepath or s3_uri.\n
+                        bucket: S3 bucket name\n
+                        filepath: full path with filname. Eg. dir1/subdir/filename.pkts\n
+                        s3_uri: full s3 URI. Eg. s3://bucket-name/filepath/filename.pkts
+                        '''
+        return http_response(status_code=400, body=response_body)
+
+    elif 'filepath' in event['queryStringParameters'] and \
+        'bucket' in event['queryStringParameters']:
+
+        bucket = event['queryStringParameters']['bucket']
+        filepath = event['queryStringParameters']['filepath']
+
+    elif 's3_uri' in event['queryStringParameters']:
+        # parse s3 uri to get bucket and filepath
+        # Eg. s3://bucket-name/filepath/filename.pkts
+        s3_uri = event['queryStringParameters']['s3_uri']
+
+        if "s3://" not in s3_uri:
+            response_body = 'Not valid S3 URI. Example input: s3://bucket/path/file.ext'
+            return http_response(status_code=400, body=response_body)
+        # Parse by '//', then parse by first occurence of '/'. Result would look like:
+        # ['bucket-name', 'filepath/filename.pkts']
+        parsed_list = s3_uri.split('//')[1].split('/', 1)
+        bucket = parsed_list[0]
+        filepath = parsed_list[1]
+
+    else:
+        response_body = '''Did not find bucket and filepath or s3_uri.\n
```

Comment:
if the bucket/filepath elif is able to be removed then this response can be updated as well

Reply:
right.